trino-aws-proxy/src/test/java/io/trino/aws/proxy/server/testing/TestingTrinoAwsProxyServer.java

```diff
@@ -208,6 +209,12 @@ public Builder withOpaContainer()
             return this;
         }
 
+        public Builder withS3RequestRewriter(Class<? extends S3RequestRewriter> impl)
```

Comment:
Instead of this could we use the SPI function from `TrinoAwsProxyServerBinding` to build a module that binds a `S3RequestRewrite` implementation?